pkg/server/updates_test.go

```diff
@@ -588,65 +476,63 @@ func TestReportUsage(t *testing.T) {
 			}
 		}
 
-		r.Lock()
-		defer r.Unlock()
-
-		if expected, actual := expectedUsageReports, r.requests; expected != actual {
+		if expected, actual := expectedUsageReports, r.NumRequests(); expected != actual {
 			t.Fatalf("expected %v reports, got %v", expected, actual)
 		}
-		if expected, actual := ts.ClusterID().String(), r.last.uuid; expected != actual {
+		last := r.LastRequestData()
+		if expected, actual := ts.ClusterID().String(), last.UUID; expected != actual {
 			return errors.Errorf("expected cluster id %v got %v", expected, actual)
 		}
-		if expected, actual := ts.node.Descriptor.NodeID, r.last.Node.NodeID; expected != actual {
+		if expected, actual := ts.node.Descriptor.NodeID, last.Node.NodeID; expected != actual {
```

Comment:
is there maybe a way to pretty-print the report in the datadriven test, and make all these assertions in the datadriven test too?